File: app/src/main/java/com/jidesh/erosnowjidesh/ViewHolders/PageViewModel.java
package com.jidesh.erosnowjidesh.ViewHolders;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.jidesh.erosnowjidesh.Models.Entities.Fav_movies;
import com.jidesh.erosnowjidesh.repository.Fav_repository;
import com.jidesh.erosnowjidesh.Models.Entities.Movie;

import java.util.List;

public class PageViewModel extends AndroidViewModel {
    private Fav_repository repository;
    private LiveData<List<Fav_movies>> allFavMovies;
    private MutableLiveData<List<Movie>> mutableLiveData;
    public PageViewModel(@NonNull Application application) {
        super(application);
        repository=new Fav_repository(application);
        allFavMovies=repository.getAllFavs();


    }
    public LiveData<List<Movie>> getSearch(String s) {

            mutableLiveData = new MutableLiveData<List<Movie>>();

          mutableLiveData=  repository.getNews(s);

        return mutableLiveData;
    }

    public void inset(Fav_movies fav_movies)
    {
        repository.Insert(fav_movies);
    }
    public void delete(Fav_movies fav_movies)
    {
        repository.Delete(fav_movies);
    }
    public void search(String search){repository.getNews(search);}
    public Integer check(int a)
    {
      return   repository.find(a);

    }

}

File: app/src/main/java/com/jidesh/erosnowjidesh/ViewHolders/PageViewModel.kt
package com.jidesh.erosnowjidesh.ViewHolders

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.jidesh.erosnowjidesh.Models.Entities.Fav_movies
import com.jidesh.erosnowjidesh.Models.Entities.Movie
import com.jidesh.erosnowjidesh.repository.Fav_repository

class PageViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: Fav_repository
    private val allFavMovies: LiveData<List<Fav_movies>>?
    private var mutableLiveData: MutableLiveData<List<Movie>>? = null


    init {
        repository = Fav_repository(application)
        allFavMovies = repository.allFavs
    }
    fun getSearch(s: String?): LiveData<List<Movie>> {
        mutableLiveData = MutableLiveData()
        mutableLiveData = repository.getNews(s!!)
        return mutableLiveData!!
    }

    fun inset(fav_movies: Fav_movies?) {
        repository.Insert(fav_movies)
    }

    fun delete(fav_movies: Fav_movies?) {
        repository.Delete(fav_movies)
    }

    fun search(search: String?) {
        repository.getNews(search!!)
    }

    fun check(a: Int): Int {
        return repository.find(a)
    }

}